addons/sale_project/models/sale_order.py

```diff
@@ -48,10 +48,18 @@ def _compute_show_project_and_task_button(self):
             ('order_id.state', 'not in', ['draft', 'sent']),
             ('product_id.detailed_type', '=', 'service'),
         ], aggregates=['order_id:array_agg'])[0][0]
+        get_create_project_button = self.env['sale.order.line']._read_group([
+            ('order_id', 'in', self.ids),
+            ('order_id.state', 'not in', ['draft', 'sent']),
+            ('product_id.detailed_type', '=', 'service'),
+        ], aggregates=['id:recordset'])[0][0]
+        is_create_project_button = True
+        for sale_order in get_create_project_button:
+            is_create_project_button = sale_order.product_template_id.service_policy in ['delivered_timesheet', 'delivered_milestones']
         for order in self:
             order.show_project_button = order.id in show_button_ids and order.project_count
             order.show_task_button = order.show_project_button or order.tasks_count
-            order.show_create_project_button = is_project_manager and order.id in show_button_ids and not order.project_count
+            order.show_create_project_button = is_project_manager and order.id in show_button_ids and not order.project_count and is_create_project_button
```

Comment:
Can't we not just do this ? 
```suggestion
            order.show_create_project_button = is_project_manager and order.id in show_button_ids and not order.project_count and order.product_template_id.service_policy in ['delivered_timesheet', 'delivered_milestones']
```
you can do the final check in a separate line if you want it to be clearer